### PR TITLE
Spark Jupyter getting started docker compose

### LINKED ISSUE #110: [FEATURE REQUEST] Quick Start: Spark via docker
### Is your feature request related to a problem? Please describe.

Not a problem. Enhancements to the "Quick Start" guide.

### Describe the solution you'd like

Include Spark docker container instead of cloning the Spark GitHub repo as described in the "Quick Start" guide

### Describe alternatives you've considered

Possibly use PySpark or other engines (Trino, PyIceberg, etc)

### Additional context

Modelled after the [Spark and Iceberg Quickstart](https://iceberg.apache.org/spark-quickstart/#spark-and-iceberg-quickstart) guide, which is defined in https://github.com/tabular-io/docker-spark-iceberg/tree/main

### PR DESCRIPTION
**Pull request by kevinjqliu**
# Description

This PR moves the `docker-compose-jupyter.yml` file (and the `notebooks/` directory), formerly in the top-level directory, into the `getting-started/spark/` folder. 

The purpose is to unify the "getting started" guides into the same directory. 

Fixes #110

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Documentation update
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

```
docker-compose -f getting-started/spark/docker-compose.yml up
```
Open the `SparkPolaris.ipynb` Jupyter notebook
Grab the `root principal credentials` from the Polaris service and replace in the notebook cell.
Run all cells in notebook

# Checklist:

Please delete options that are not relevant.

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] If adding new functionality, I have discussed my implementation with the community using the linked GitHub issue

### CODE REVIEW

**Comment by kevinjqliu** [comment]
I want to make sure this is something we want to do before proceeding to add more to the PR

cc @collado-mike / @flyrain

**Comment by flyrain** [comment]
Make sense to me. Thanks @kevinjqliu! Do we have any doc for its usage? We may add doc if not.

**Comment by kevinjqliu** [comment]
@flyrain yep i'll have a README in here, similar to the trino one

**Comment by flyrain** [comment]
Sounds good. We will need these doc to be in the Polaris doc site, like this https://polaris.apache.org/docs/overview/. I couldn't find Trino's doc there, this may involve doc publish and link. cc @jbonofre

**Comment by kevinjqliu** [comment]
I see, this is the [README for trino](https://github.com/apache/polaris/blob/main/getting-started/trino/README.md). I'll add a similar README for spark. 

As a follow-up, we can change the Polaris doc to refer to these guides https://polaris.apache.org/docs/quickstart

**Comment by collado-mike** [comment]
This looks good to me. We should change the name of the compose file to just `docker-compose.yml` so we don't have to specify the filename in the command line :)

**Comment by kevinjqliu** [comment]
@collado-mike makes sense, will do. 

I have a question [on slack](https://apache-iceberg.slack.com/archives/C07HADW86HZ/p1726508953511979) about unable to assume the role `arn:aws:iam::631484165566:role/datalake-storage-integration-role` in the notebook, do you mind taking a look?

File: getting-started/spark/create-polaris-catalog.sh
Comment (sfc-gh-emaynard) [inline getting-started/spark/create-polaris-catalog.sh:1]: I wonder if it might be easier to use the CLI here

Comment (kevinjqliu) [reply getting-started/spark/create-polaris-catalog.sh:1]: might be, if you want a spark-shell. I think the jupyter notebook does a good job of explaining a lot of the concepts

File: getting-started/spark/README.md
Comment (kevinjqliu) [reply getting-started/spark/README.md:1]: yes I'll admit, this README is a filler for now as a way to get spark & polaris up and running quickly

File: getting-started/spark/README.md
Comment (kevinjqliu) [reply getting-started/spark/README.md:37]: which part is the typo?

File: getting-started/spark/create-polaris-catalog.sh
Comment (sfc-gh-emaynard) [reply getting-started/spark/create-polaris-catalog.sh:1]: Sorry, I meant the `polaris` CLI instead of using `curl`

File: getting-started/spark/README.md
Comment (sfc-gh-emaynard) [reply getting-started/spark/README.md]: The existing text just says `to access Jupyter notebook`

File: getting-started/spark/create-polaris-catalog.sh
Comment (kevinjqliu) [reply getting-started/spark/create-polaris-catalog.sh:1]: ah i dont know how to use the polaris CLI, so i just copied directly from https://github.com/apache/polaris/blob/main/regtests/run_spark_sql.sh

File: getting-started/spark/README.md
Comment (kevinjqliu) [reply getting-started/spark/README.md]: changed to `to access the Jupyter notebook`

**Comment by kevinjqliu** [comment]
md check intermittently shows `https://redocly.com/docs/cli/installation` as 400 error, weird

**Comment by flyrain** [comment]
> md check intermittently shows https://redocly.com/docs/cli/installation as 400 error, weird

It's OK to remove the link for now since we’re transitioning to Hugo.

**Comment by kevinjqliu** [comment]
@flyrain just had to run the CI a few times, it's unrelated to this change

File: .github/workflows/check-md-link.yml
Comment (kevinjqliu) [inline .github/workflows/check-md-link.yml:44]: This PR moved `notebooks/` from top-level directory into the `getting-started/` directory

File: getting-started/spark/create-polaris-catalog.sh
Comment (flyrain) [inline getting-started/spark/create-polaris-catalog.sh:23]: I'm not entirely sure if we need this file. Could we handle everything directly within the notebook, like the other operations in `SparkPolaris.ipynb`? Would it simplify things if we moved the operations there?

File: getting-started/spark/README.md
Comment (flyrain) [inline getting-started/spark/README.md:22]: There is other way to try Spark with Polaris without docker, it's a not a blocker, we can add it later.

**Review by flyrain — COMMENTED** [review]
Thanks @kevinjqliu for working on it. LGTM overall. Left some comments and questions.

File: getting-started/spark/create-polaris-catalog.sh
Comment (kevinjqliu) [reply getting-started/spark/create-polaris-catalog.sh:23]: we could, but i think its a good idea to separate infra code (this script) from application code (the notebook)

**Comment by kevinjqliu** [comment]
Thanks for the review @flyrain, addressed your comments

File: getting-started/spark/create-polaris-catalog.sh
Comment (flyrain) [reply getting-started/spark/create-polaris-catalog.sh:23]: We could initialize catalog in notebook as well. I feel it's more flexible that way. for example, you don't have to worry about the an env variable for catalog name. But I'm OK with either one. Not a blocker for me.

**Review by flyrain — APPROVED** [review]

**Comment by flyrain** [comment]
We cannot merge any PR until #374 is merged.

**Comment by kevinjqliu** [comment]
Thanks for the heads up, I'll rebase once that PR's merged

**Comment by kevinjqliu** [comment]
@flyrain took your advice, moved `getting-started/spark/create-polaris-catalog.sh` logic into the jupyter notebook. 
Also rebased off latest main. I think this PR is good to go. Please take a look!

**Review by flyrain — APPROVED** [review]

**Comment by flyrain** [comment]
Thanks a lot for working on it, @kevinjqliu! Thanks all for the review.

**Pull request merged**
